sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -46,12 +46,13 @@ def test_n_features_to_select(direction, n_features_to_select):
     assert sfs.transform(X).shape[1] == n_features_to_select
 
 
-@pytest.mark.parametrize("direction", ("forward", "backward"))
-def test_n_features_to_select_auto(direction):
-    """Check the behaviour of `n_features_to_select="auto"` with different
-    values for the parameter `tol`.
+@pytest.mark.parametrize(
+    "direction,max_features_to_select", (("forward", 9), ("backward", 10))
+)
+def test_n_features_to_select_auto(direction, max_features_to_select):
+    """Check the behaviour of `n_features_to_select="auto"` when selecting
+    features in a forward and backward direction.
     """
-
     n_features = 10
     tol = 1e-3
```

Comment:
Why does a positive `tol` work when doing backwards selection?

Reply:
The answer is that the sign of `tol` is only checked if the selection is `"forward"`. Something to improve maybe.